homeassistant/components/calendar/__init__.py

```diff
@@ -156,23 +162,26 @@ def _get_date(date):
 
         summary = self.data.event.get('summary', '')
 
-        # check if we have an offset tag in the message
+        # check if we have an offset tag in the message of the event
         # time is HH:MM or MM
         reg = '{}([+-]?[0-9]{{0,2}}(:[0-9]{{0,2}})?)'.format(self._offset)
         search = re.search(reg, summary)
         if search and search.group(1):
             time = search.group(1)
-            if ':' not in time:
-                if time[0] == '+' or time[0] == '-':
-                    time = '{}0:{}'.format(time[0], time[1:])
-                else:
-                    time = '0:{}'.format(time)
+        else:   # If not, we check the calendar value, which defaults to 0
```

Comment:
I think parsing the time manually here is somewhat of a hack, which is probably necessary when retrieving the offset time from the event title, but not if we get the offset from the config. For the config value, we can use the neat `time_period` validator from `config_validation.py`, which then allows us to specify the time interval in the config either as a string, a number of seconds, or a map. 

So, my suggestion: Move lines 174 to 180 into the `if search and search.group(1):` block, and do 
```
else:
  offset_time = self._offset_time
```
(requiring `self._offset_time` to already be a timedelta).